From 79482b13b67de99094268b230a6027e99d8296dd Mon Sep 17 00:00:00 2001
From: Hans Rosenfeld <hans.rosenfeld@joyent.com>
Date: Thu, 18 May 2017 14:48:08 +0200
Subject: [PATCH] OS-6072 increase ramdisk size by 3M for DEBUG kernels
 Reviewed by: Robert Mustacchi <rm@joyent.com> Reviewed by: Jerry Jelinek
 <jerry.jelinek@joyent.com> Reviewed by: Joshua M. Clulow <jmc@joyent.com>
 Approved by: Jerry Jelinek <jerry.jelinek@joyent.com>

---
 tools/build_live | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/tools/build_live b/tools/build_live
index cca886a8..77564714 100755
--- a/tools/build_live
+++ b/tools/build_live
@@ -65,6 +65,8 @@ bi_file_usr=
 bi_mnt_root=
 bi_mnt_usr=
 
+bi_root_size=
+
 bi_log=
 
 
@@ -786,7 +788,15 @@ bi_setup_work_dir
 # Create and mount the "/" (root) ramdisk image:
 #
 bi_file_root="$bi_tmpdir/root.image"
-bi_create_ramdisk 272000 "$bi_file_root"
+bi_root_size=272000
+if strings "$bi_wsroot/proto/kernel/amd64/genunix" |
+    grep "DEBUG enabled" >/dev/null; then
+	#
+	# Increase root size by 3M if we have a DEBUG kernel.
+	#
+	bi_root_size=$(( bi_root_size + 3000))
+fi
+bi_create_ramdisk "$bi_root_size" "$bi_file_root"
 bi_lofi_root="$bi_lofi_last"
 bi_mnt_root="$bi_tmpdir/a"
 bi_create_ufs 12248 "$bi_lofi_root" "$bi_mnt_root" "/"
-- 
2.21.0

